snowplow-tracker/src/main/java/com/snowplowanalytics/core/tracker/ServiceProvider.kt

```diff
@@ -48,6 +49,7 @@ class ServiceProvider(
     private var subjectController: SubjectControllerImpl? = null
     private var sessionController: SessionControllerImpl? = null
     private var gdprController: GdprControllerImpl? = null
+    private var ecommerceController: EcommerceControllerImpl? = null
```

Comment:
This is a bit of a weird Java style dance that we do here with the `getOrMakeXY` and `makeXY` methods in addition to the `XY` property. I think we can achieve the same thing by making the property with lazy initialization – see the `pluginsController` below.